dipy/workflows/stats.py

```diff
@@ -0,0 +1,144 @@
+            Where the resulting file will be saved. (default '')
+        out_file : string, optional
+            Name of the result file to be saved. (default 'product.json')
+        """
+        
+        if not isinstance(bbox_threshold, tuple):
+            b = bbox_threshold.replace("[","")
+            b = b.replace("]","")
+            b = b.replace("(","")
+            b = b.replace(")","")
+            b = b.replace(" ","")
+            b = b.split(",")
+            for i in range(len(b)):
+                b[i] = float(b[i])
+            bbox_threshold = tuple(b)
+
+        io_it = self.get_io_iterator()
+        
+        for data_path, data_bvals_path, data_bvecs_path, out_path in io_it:
+           
+            img = nib.load('{0}'.format(data_path))
+            bvals, bvecs = read_bvals_bvecs('{0}'.format(data_bvals_path), '{0}'.format(data_bvecs_path))
+            gtab = gradient_table(bvals, bvecs)
+            
+            data = img.get_data()
+            affine = img.affine
+
+            if mask == None:
+                logging.info('Computing brain mask...')
+                b0_mask, mask = median_otsu(data)
+            else:
+                mask = nib.load(mask).get_data().astype(np.bool)
+
+            logging.info('Computing tensors...')
+            tenmodel = TensorModel(gtab)
+            tensorfit = tenmodel.fit(data, mask=mask)
+
+            logging.info('Computing worst-case/best-case SNR using the corpus callosum...')
+            threshold = bbox_threshold
+            
+            if np.ndim(data) == 4:
+                CC_box = np.zeros_like(data[..., 0])
+            elif np.ndim(data) == 3:
+                CC_box = np.zeros_like(data)
+            else:
+                raise IOError('DWI data has invalid dimensions')
+            
+            mins, maxs = bounding_box(mask)
+            mins = np.array(mins)
+            maxs = np.array(maxs)
+            diff = (maxs - mins) // 4
+            bounds_min = mins + diff
+            bounds_max = maxs - diff
+
+            CC_box[bounds_min[0]:bounds_max[0],
+                   bounds_min[1]:bounds_max[1],
+                   bounds_min[2]:bounds_max[2]] = 1
+                   
+            mask_cc_part, cfa = segment_from_cfa(tensorfit, CC_box, threshold,
+                                                 return_cfa=True)
+                                                 
+            cfa_img = nib.Nifti1Image((cfa*255).astype(np.uint8), affine)
+            mask_cc_part_img = nib.Nifti1Image(mask_cc_part.astype(np.uint8), affine)
+            nib.save(mask_cc_part_img, 'cc.nii.gz')
+
+            mean_signal = np.mean(data[mask_cc_part], axis=0)
+            mask_noise = binary_dilation(mask, iterations=10)
+            mask_noise[..., :mask_noise.shape[-1]//2] = 1
+            mask_noise = ~mask_noise
+            mask_noise_img = nib.Nifti1Image(mask_noise.astype(np.uint8), affine)
+            nib.save(mask_noise_img, 'mask_noise.nii.gz')
```

Comment:
It seems you save 2 images (`cc.nii.gz` and `mask_noise.nii.gz`) so it should be an output.